src/view/domconverter.js

```diff
@@ -243,22 +247,41 @@ export default class DomConverter {
 	 * @returns {Iterable.<Node>} DOM nodes.
 	 */
 	* viewChildrenToDom( viewElement, domDocument, options = {} ) {
-		const fillerPositionOffset = viewElement.getFillerOffset && viewElement.getFillerOffset();
+		const blockFillerOffset = viewElement.getFillerOffset && viewElement.getFillerOffset();
+		const inlineFillerOffset = options.inlineFiller && options.inlineFiller.parent == viewElement ? options.inlineFiller.offset : null;
 		let offset = 0;
 
 		for ( const childView of viewElement.getChildren() ) {
-			if ( fillerPositionOffset === offset ) {
+			if ( blockFillerOffset === offset ) {
 				yield this.blockFiller( domDocument );
 			}
 
-			yield this.viewToDom( childView, domDocument, options );
+			const domNode = this.viewToDom( childView, domDocument, options );
+
+			// We need to insert inline filler before the dom node.
+			if ( inlineFillerOffset === offset ) {
+				// If dom node that was about to return is a text node, just add inline filler to that text node.
+				if ( this.isText( domNode ) ) {
+					domNode.data = INLINE_FILLER + domNode.data;
```

Comment:
I'd be worried that the renderer may not expect to get the filler in some places. Especially that it needs to track it in `Renderer#_inlineFiller` – so how will it do this? I see that `_getInlineFillerNode()` should handle this but since we're leaking this logic to a separate class we'll risk that this will desync.